compiler/packages/babel-plugin-react-compiler/src/Inference/InferEffectDependencies.ts

```diff
@@ -189,9 +190,10 @@ export function inferEffectDependencies(fn: HIRFunction): void {
              */
             for (const dep of scopeInfo.deps) {
               if (
-                (isUseRefType(dep.identifier) ||
+                ((isUseRefType(dep.identifier) ||
                   isSetStateType(dep.identifier)) &&
-                !reactiveIds.has(dep.identifier.id)
+                  !reactiveIds.has(dep.identifier.id)) ||
+                isFireFunctionType(dep.identifier)
```

Comment:
Ahh makes sense, fire is different than setState / refs in that we always codegen the `useFire` hook call, so there is no "reactive reference to non-reactive values" issue (e.g. `props.cond ? ref1 : ref2`)